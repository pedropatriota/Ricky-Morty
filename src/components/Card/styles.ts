import styled from 'styled-components';

export const Container = styled.div`
	display: flex;
	flex-direction: column;
	width: 18rem;
`;

export const ImagContainer = styled.div<{ image: string }>`
	background-image: url(${props => props.image});
	background-repeat: no-repeat;
	background-size: cover;
	width: 100%;
	max-width: 18rem;
	height: 18rem;
	border-radius: 1rem 1rem 0 0;
`;

export const InfoContainer = styled.div`
	padding: 0.5rem;
	background-color: ${props => props.theme.colors.el};
	border: 1px solid ${props => props.theme.colors.el};
	border-radius: 0 0 1rem 1rem;

	h2 {
		margin-bottom: 1rem;
		color: ${props => props.theme.colors.text};
	}
	p {
		margin-bottom: 0.4rem;
		color: ${props => props.theme.colors.text};
	}
`;
