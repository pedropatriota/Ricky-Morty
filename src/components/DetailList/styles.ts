import styled from 'styled-components';

export const Container = styled.div`
	display: flex;
	gap: 0.75rem;
	flex-direction: column;
	margin-bottom: 0.75rem;
`;
