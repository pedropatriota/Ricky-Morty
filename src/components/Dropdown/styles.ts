import styled from 'styled-components';
import Select from 'react-select';

export const ReactSelect = styled(Select)`
  .react-select__control {
    width: min(90vw, 12.5rem);
    background-color: ${(props) => props.theme.colors.el};
    border-color: ${(props) => props.theme.colors.border};
    box-shadow: 0 0 0 transparent;
    height: 3.375rem;

    &:hover {
      border-color: ${(props) => props.theme.colors.border};
    }
  }

  .react-select__single-value {
    color: ${(props) => props.theme.colors.text};
  }

  .react-select__menu {
    width: 12.5rem;
    background-color: ${(props) => props.theme.colors.el};
    color: ${(props) => props.theme.colors.text};
  }

  .react-select__option,
  .react-select__option--is-focused 
  {
    background-color: ${(props) => props.theme.colors.el};
    color: ${(props) => props.theme.colors.text};
  }

  .react-select__dropdown-indicator,
  .react-select__indicator-separator {
    display: none;
  }
`;
