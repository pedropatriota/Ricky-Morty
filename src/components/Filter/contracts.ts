import { ActionMeta, GroupBase, SingleValue } from 'react-select';

export type TFilter = SingleValue<{ label: string; value: string }>;

export type TFilterFN = (
	newValue: TFilter | unknown,
	actionMeta: ActionMeta<TFilter | unknown>
) => void;

export interface IFilterProps {
	filter: TFilter[];
	options: readonly (TFilter | GroupBase<{ label: string; value: string }>)[][];
	label: string[];
	handleFilter: TFilterFN[];
}
