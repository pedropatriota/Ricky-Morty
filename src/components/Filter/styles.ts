import styled from 'styled-components';

export const Container = styled.div`
	display: flex;
	align-items: flex-end;
	justify-content: space-between;
	flex-wrap: wrap;
	gap: 0.8rem;

	@media (max-width: 500px) {
		flex-wrap: wrap;
		gap: 1rem;
	}
`;
