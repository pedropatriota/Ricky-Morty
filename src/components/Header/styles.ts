import styled from 'styled-components';

export const Header = styled.header`
	padding: 0px min(5vw, 5rem);
	background-color: ${({ theme }) => theme.colors.bg};
	box-shadow: 0px 10px 15px -3px rgba(0, 0, 0, 0.1);
	margin-bottom: 4rem;
`;

export const Container = styled.div`
	display: flex;
	justify-content: space-between;
	align-items: center;
	flex-wrap: wrap;

	img {
		width: 100%;
		height: auto;
		max-width: 10rem;
	}
`;
