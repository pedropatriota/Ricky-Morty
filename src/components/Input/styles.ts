import styled from 'styled-components';

export const Input = styled.input`
  width: min(90vw, 30rem);
  border: ${(props) => props.theme.colors.border};
  color: ${(props) => props.theme.colors.text};
  height: 3.375rem;
  border-radius: 3px;
  padding: 0 2rem;
  font-size:0.94rem;
  outline: none;
  background: ${(props) => props.theme.colors.el}
`;
