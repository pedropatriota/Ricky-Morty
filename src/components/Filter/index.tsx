import Dropdown from '../Dropdown';
import type { IFilterProps } from './contracts';
import Input, { IFormProps } from '../Input';
import { Container } from './styles';

const Filter = ({
	value,
	handleChange,
	handleFilter,
	filter,
	options,
	placeholder
}: IFilterProps & IFormProps) => {
	return (
		<Container>
			<Input
				value={value}
				handleChange={handleChange}
				placeholder="type a name..."
			/>
			<Dropdown
				filter={filter[0]}
				options={options[0]}
				handleFilter={handleFilter[0]}
				placeholder={placeholder[0]}
			/>
			<Dropdown
				filter={filter[1]}
				options={options[1]}
				handleFilter={handleFilter[1]}
				placeholder={placeholder[1]}
			/>
		</Container>
	);
};

export default Filter;
