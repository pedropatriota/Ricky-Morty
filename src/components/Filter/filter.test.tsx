import { render, screen, fireEvent } from '@testing-library/react';
import Filter from '.';
import type {} from './contracts';
import { ThemeProvider } from 'styled-components';
import { light } from '../../styles/themes';

describe('Filter', () => {
	const props = {
		value: 'Rick',
		handleChange: jest.fn(),
		handleFilter: [jest.fn(), jest.fn()],
		filter: [
			{ label: '', value: '' },
			{ label: '', value: '' }
		],
		options: [
			[
				{ label: 'Female', value: 'Female' },
				{ label: 'Male', value: 'Male' },
				{ label: 'Unknown', value: 'unknown' }
			],
			[
				{ label: 'Alive', value: 'Alive' },
				{ label: 'Dead', value: 'Dead' },
				{ label: 'Unknown', value: 'unknown' }
			]
		],
		label: ['gender', 'status']
	};

	it('should render correctly', () => {
		render(
			<ThemeProvider theme={light}>
				<Filter {...props} />
			</ThemeProvider>
		);

		const input = screen.getByRole('textbox');
		const select = screen.getAllByLabelText(/select/i) as HTMLElement[];

		expect(input).toBeInTheDocument();
		expect(select).toHaveLength(2);
	});

	it('should change the input value', () => {
		render(
			<ThemeProvider theme={light}>
				<Filter {...props} />
			</ThemeProvider>
		);

		const input = screen.getByRole('textbox');
		fireEvent.click(input);
		fireEvent.change(input, { target: { value: 'Beth' } });

		expect(props.handleChange).toBeCalledTimes(1);
	});

	it('should select an option', async () => {
		render(
			<ThemeProvider theme={light}>
				<Filter {...props} />
			</ThemeProvider>
		);

		const selector = (await screen.findByLabelText(/gender/)) as Element;

		fireEvent.keyDown(selector, {
			key: 'ArrowDown',
			keyCode: 40,
			code: 40
		});

		const selectedItem = screen.getAllByText('Female');

		fireEvent.click(selectedItem[0]);

		expect(props.handleFilter[0]).toBeCalled();
	});

	it('should match the snapshot', () => {
		const { container } = render(
			<ThemeProvider theme={light}>
				<Filter {...props} />
			</ThemeProvider>
		);
		expect(container).toMatchSnapshot();
	});
});
