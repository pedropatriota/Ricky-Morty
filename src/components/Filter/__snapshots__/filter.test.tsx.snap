// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter should match the snapshot 1`] = `
<div>
  <div
    class="sc-eqUAAy eHNHHl"
    data-testid="filter"
  >
    <input
      class="sc-gEvEer bANnIj"
      placeholder="type a name..."
      type="text"
      value="Rick"
    />
    <div>
      <label
        for="react-select-2-input"
      >
        Select a gender:
      </label>
      <div
        class="sc-aXZVg bEtCNO css-b62m3t-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-8-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class="react-select__control css-13cymwt-control"
        >
          <div
            class="react-select__value-container react-select__value-container--has-value css-1fdsijx-ValueContainer"
          >
            <div
              class="react-select__single-value css-1dimb5e-singleValue"
            />
            <div
              class="react-select__input-container css-qbdosj-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                aria-haspopup="true"
                aria-label="Select a gender:"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class="react-select__input"
                id="react-select-8-input"
                role="combobox"
                spellcheck="false"
                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
          >
            <div
              aria-hidden="true"
              class="react-select__indicator react-select__clear-indicator css-1xc3v61-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                />
              </svg>
            </div>
            <span
              class="react-select__indicator-separator css-1u9des2-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class="react-select__indicator react-select__dropdown-indicator css-1xc3v61-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <label
        for="react-select-2-input"
      >
        Select a status:
      </label>
      <div
        class="sc-aXZVg bEtCNO css-b62m3t-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-9-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class="react-select__control css-13cymwt-control"
        >
          <div
            class="react-select__value-container react-select__value-container--has-value css-1fdsijx-ValueContainer"
          >
            <div
              class="react-select__single-value css-1dimb5e-singleValue"
            />
            <div
              class="react-select__input-container css-qbdosj-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                aria-haspopup="true"
                aria-label="Select a status:"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class="react-select__input"
                id="react-select-9-input"
                role="combobox"
                spellcheck="false"
                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
          >
            <div
              aria-hidden="true"
              class="react-select__indicator react-select__clear-indicator css-1xc3v61-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                />
              </svg>
            </div>
            <span
              class="react-select__indicator-separator css-1u9des2-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class="react-select__indicator react-select__dropdown-indicator css-1xc3v61-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
